src/components/button/button.tsx

```diff
@@ -200,7 +200,16 @@ export class Button
   }
 
   componentDidLoad(): void {
+    const { initialSlottedText, el } = this;
     setComponentLoaded(this);
+
+    const spanContent = el.shadowRoot.querySelector("span.content");
+    const spanContentOffsetWidth = (spanContent as HTMLElement).offsetWidth;
+    const button = el.shadowRoot.querySelector("button");
```

Comment:
Instead of querying for these nodes, can you safe these nodes using the `ref` property on them?